### ISSUE TITLE
Errors due to forecast duration (N2) for data after the 18/04/2020

### ISSUE THREAD

**Issue by payoto** [ImperialCollegeLondon/covid19model#77]
**Describe the bug**

When running with data including and after the 18/04/20 the plot-forecast.r fail to run.

**Diagnostic** 

This is because, at that stage Italy has 84 days of data, the forecasting is attempted for 7 days (hard coded in the forecast), and the model is only run for 90 days `N2` in `base.r`.

**The fixes**

After a bit of poking around to find a fix:

- In `base.r` increase the value of `N2` to 100 (its max allowed) This fix works for 10 days.
- Increase the number of values in file [`data/serial_interval.csv`](data/serial_interval.csv)
  - should this just be padded with zeros? I'm not sure what those do.

**To Reproduce**

1. Run `Rscript data/fetch-ecdc.r`
2. Run `Rscript base.r base --debug`
6. Inspect closely the output (this does not appear in the CI runs as they don't monitor the status of intermediate steps)

**Error message**

```
Saving 10 x 7 in image
Error in estimated.deaths[, 1:N2, i] : subscript out of bounds
Calls: make_forecast_plot -> colMeans -> is.data.frame
In addition: Warning messages:
1: Transformation introduced infinite values in continuous y-axis
2: Removed 27 rows containing missing values (geom_bar).
3: Transformation introduced infinite values in continuous y-axis
4: Removed 27 rows containing missing values (geom_bar).
5: Transformation introduced infinite values in continuous y-axis
6: Removed 27 rows containing missing values (geom_bar).
7: Transformation introduced infinite values in continuous y-axis
8: Removed 27 rows containing missing values (geom_bar).
Execution halted
[1] "Running results/base-414043-stanfit.Rdata"
Error in prediction[, 1:length(x), i] : subscript out of bounds
Calls: colMeans -> is.data.frame
Execution halted
```

**Comment by s-mishra** [comment]
Hi @payoto, yeah I was worried about this. Anyways we are soon releasing a new version with the ability to do forecasts for much longer. Importantly we will remove N2 to be a pre-defined value. Just give us some time as we are in middle of a busy period.

**Comment by payoto** [comment]
Hi @s-mishra no worries, thanks for the quick answer, If you have time could you comment on the validity of applying 0s for longer forecast periods?

**Comment by s-mishra** [comment]
Solved by merging your PR and new commit too 865806ff6f7afb393
